Library/TelemetryClient.ts

```diff
@@ -217,6 +218,15 @@ class TelemetryClient {
             }
         }
 
+        // Sanitize tags and properties after running telemetry processors
+        if (accepted) {
+            envelope.tags = Util.validateStringMap(envelope.data.baseData.properties) as Tags & Tags[];
```

Comment:
This sanitization will be triggered for all events even if no telemetry processors and tags were added, in the case of properties will be executed twice always, I wonder if this could be fixed having correct types in typescript preventing this to happen in the first place
https://github.com/microsoft/ApplicationInsights-node.js/blob/develop/Declarations/Contracts/TelemetryTypes/EnvelopeTelemetry.ts#L64

Reply:
Hm... I intended for this to only run when Telemetry processors are attached. I wouldn't want others to take the performance hit.

Unfortunately, for those of us who don't use type script (or have dynamic inputs at runtime) is that this a non-string value fails silently and users are unlikely to have `setInternalLogging` on. Customer support isn't trained on this scenario and it took months to resolve and we figured it out in-house.

Rather than iterating over the map it might be cheaper to have the tags/properties objects use getters and setters that casts to strings passed to the telemetry processors.

Another place where this type of silent conversion happens is `process.env`; so it's not unheard of.

Reply:
@hectorhdzg: it looks like `runTelemetry Processors` exits early if none are attached?

Reply:
You are right, I got confused github file changes view didn't help this time